test/unit/autolinker_spec.js

```diff
@@ -0,0 +1,220 @@
+/* Copyright 2025 Mozilla Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Autolinker } from "../../web/autolinker.js";
+
+describe("autolinker", function () {
+  it("should correctly find URLs", function () {
+    const matches = Autolinker.findLinks("http://www.example.com");
+    expect(matches[0].url).toEqual("http://www.example.com/");
+  });
+
+  it("should correctly find simple valid URLs", function () {
+    const links = [
+      [
+        "http://subdomain.example.com/path/to/page?query=param",
+        "http://subdomain.example.com/path/to/page?query=param",
+      ],
+      [
+        "www.example.com/path/to/resource",
+        "http://www.example.com/path/to/resource",
+      ],
+      [
+        "http://example.com/path?query=value#fragment",
+        "http://example.com/path?query=value#fragment",
+      ],
+    ];
+    const matches = Autolinker.findLinks(links.map(link => link[0]).join("\n"));
+    expect(matches.length).toEqual(3);
```

Comment:
nit: here and elsewhere in the tests, could you use a loop instead of unrolling it ?

Reply:
done!